docs/source/best-practices/checkpointing.rst

```diff
@@ -110,45 +167,67 @@ Where the checkpointing implementation is given in this code snippet of ``train_
 
 To submit this job to the scheduler, use the command::
-   
-  sbatch submit_tf_array.bash
-
-Checkpointing using DMTCP
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-`DMTCP <https://dmtcp.sourceforge.io/>`_ (Distributed MultiThreaded checkpointing) is a checkpointing tool that lets you checkpoint without the need to change your code. It Works with most Linux applications such as Python, Matlab, R, GUI, MPI etc. 
-The program runs in the background of your program, without significant performance loss, and saves the process states into checkpoint files. DMTCP is available on the cluster ::
-
- module avail dmtcp
- module show dmtcp
- module load dmtcp/2.6.0
-
-As DMTCP runs in the background, it requires some changes to your Shell script. For examples of how to checkpoint with DMTCP visit `here <https://github.com/northeastern-rc/training-checkpointing/tree/main/Exercise_3>`_. 
-The example demonstrates how to use DMTCP with a simple C++ program (scripts modified from `RSE-Cambridge <https://github.com/RSE-Cambridge/dmtcp-tests>`_).
-
-
-Checkpointing tips
-~~~~~~~~~~~~~~~~~~~
 
-What data to save?
- * Non-temporary application data
- * Any application data that has been modified since the last checkpoint
- * Delete checkpoints that are no longer useful - keep only the most recent checkpoint file.
+  sbatch submit_tf_array.bash
 
-How frequently to checkpoint? 
- * Too often – will slow down your calculation, may be I/O heavy and memory-limited.
- * Too infrequently – leads to large/long rollback times.
- * Consider how long it takes to checkpoint and restart your calculation. 
- * In most cases a rate of every 10-15 minutes is ok.
+Python PyTorch
+------------------
+
+Tips and tricks
+------------------
+
+Save only the model's State_dict
+""""""""""""""""""""""""""""""""
+Save only the model's state_dict and the optimizer's state, as this allows us to save only the
+necessary information needed to resume training. In addition, this reduces the size of the checkpoint file and makes it
+easier to load the model.
+
+Save regularly
+""""""""""""""""""""
+To prevent losing progress in case of a crash or interruption, save the checkpoint file regularly (i.e., after each epoch).
+
+Save to multiple locations
+""""""""""""""""""""""""""""""""
+Save the checkpoint file to multiple locations, such as a local drive and the cloud, to ensure that
+the checkpoint is recovered in case of failure.
+
+Use the latest versions of libraries
+""""""""""""""""""""""""""""""""""""""""
+Using the latest version of PyTorch and other relevant libraries is vital; changes in these libraries may cause
+compatibility issues with older checkpoints. With these best practices, you can ensure that your PyTorch models are
+saved efficiently and effectively and that your progress is not lost in case of a crash or interruption.
+
+Avoid saving unnecessary information
```

Comment:
Can the information from this section be combined with the Save only the model's Sate_dict section?